main.go

```diff
@@ -97,48 +100,26 @@ func main() {
 		newWords     = 0
 	)
 
-	scanner := bufio.NewScanner(os.Stdin)
-	for scanner.Scan() {
-		target := scanner.Text()
-		if target == "" {
-			continue
-		}
+	if val := args["--input"]; val != nil {
+		args["--format"] = "json"
+		target := val.(string)
+		addNewWords(target, &nowStreak, &newWords, anki, args)
 
-		front, back, err := getFrontBack(target, args["--format"].(string))
-		if err != nil {
-			log.Fatal(err)
-		}
+	} else {
+		scanner := bufio.NewScanner(os.Stdin)
+		for scanner.Scan() {
+			target := scanner.Text()
 
-		found, err := anki.Search(front)
-		if err != nil {
-			log.Fatal(karma.Format(err, "unable to search %q", front))
-		}
-
-		if found {
-			log.Debugf("%q is already exists", front)
-
-			nowStreak++
+			addNewWords(target, &nowStreak, &newWords, anki, args)
```

Comment:
the function returns error if an error has occurred, could you handle that error here and call `log.Fatal` if it's not nil?